File: LibraryManager/Controller.py
import adminmanager
import bookmanager
import usermanager
import database

def AdminMenu():
    return """
                Admin Menu:
                0. Show admin menu
                1. Issue book
                2. Return book
                3. Add book
                4. Remove book
                5. Ban user
                6. Unban user
                7. Add admin
                8. Remove admin
                9. View all books
                10. Clear User Data
                11. View user issued books
                """

def UserMenu():
    return """
            Main Menu:
            1. View your issued books
            2. Book Reservation
            3. View books
            """

def ReservationMenu():
    return """
            Reservation Menu:
            1. Reserve a book
            2. Cancel reservation
            3. View reservation
            4. Return
            """

def Handler(msgbody, sender):
    mydbobj = database.connector()
    userauth = mydbobj.GetAuth(sender)
    print(userauth)
    if userauth == 'Admin':
        mode = adminmanager.GetAdminData('Mode', sender)
        if mode == 'AdminMenu':
            if msgbody == '1':
                adminmanager.SetUserData('Mode', sender, 'IssueBook')
                return "Enter user phone number and book ID in the following format: \nPhone_Number/Book_ID"
            elif msgbody == '2':
                adminmanager.SetUserData('Mode', sender, 'ReturnBook')
                return "Enter user phone number and book ID in the following format: \nPhone_Number/Book_ID"
            elif msgbody == '3':
                adminmanager.SetUserData('Mode', sender, 'AddBook')
                return """
                Enter Book details in the given format:
                Book_Name/Book_Genre
                """
            elif msgbody == '4':
                adminmanager.SetUserData('Mode', sender, 'RemoveBook')
                return """
                Enter the Book ID that you wish to remove:
                """
            elif msgbody == '5':
                adminmanager.SetUserData('Mode', sender, 'BanUser')
                return "Enter user phone number: "
            elif msgbody == '6':
                adminmanager.SetUserData('Mode', sender, 'UnbanUser')
                return "Enter user phone number: "
            elif msgbody == '7':
                adminmanager.SetUserData('Mode', sender, 'AddAdmin')
                return "Enter new admin details: \nPhone_Number/Name"
            elif msgbody == '8':
                adminmanager.SetUserData('Mode', sender, 'RemoveAdmin')
                return "Enter admin phone number: (+91XXXXXXXXXX)"
            elif msgbody == '9':
                return bookmanager.ViewAllBooks()
            elif msgbody == '10':
                adminmanager.SetUserData('Mode', sender, 'ClearUserData')
                return 'Enter user phone number: (+91XXXXXXXXXX)'
            elif msgbody == '11':
                adminmanager.SetUserData('Mode', sender, 'ViewUserBooksIssued')
                return 'Enter user phone number: (+91XXXXXXXXXX)'
            else:
                return AdminMenu()
        elif mode == 'IssueBook':
            num, bookid = msgbody.split('/')
            adminmanager.SetUserData('Mode', sender, 'AdminMenu')
            return usermanager.Issue(bookid, num, sender)
        elif mode == 'ReturnBook':
            num, bookid = msgbody.split('/')
            adminmanager.SetUserData('Mode', sender, 'AdminMenu')
            return usermanager.Return(bookid, num)
        elif mode == 'AddBook':
            name, genres = msgbody.split('/')
            genres_list = genres.split(' ')
            adminmanager.SetUserData('Mode', sender, 'AdminMenu')
            return bookmanager.AddBook([name, sender], genres_list)
        elif mode == 'RemoveBook':
            adminmanager.SetUserData('Mode', sender, 'AdminMenu')
            return bookmanager.RemoveBook(msgbody)
        elif mode == 'BanUser':
            adminmanager.SetUserData('Mode', sender, 'AdminMenu')
            return usermanager.BanUser(msgbody, sender)
        elif mode == 'UnbanUser':
            adminmanager.SetUserData('Mode', sender, 'AdminMenu')
            return usermanager.UnbanUser(msgbody)
        elif mode == 'AddAdmin':
            adminnum, name = msgbody.split('/')
            adminmanager.SetUserData('Mode', sender, 'AdminMenu')
            return adminmanager.AddAdmin([adminnum, name])
        elif mode == 'RemoveAdmin':
            adminmanager.SetUserData('Mode', sender, 'AdminMenu')
            return adminmanager.RemoveAdmin(msgbody)
        elif mode == 'ClearUserData':
            adminmanager.SetUserData('Mode', sender, 'AdminMenu')
            return usermanager.RemoveUser(msgbody)
        elif mode == 'ViewUserBooksIssued':
            adminmanager.SetUserData('Mode', sender, 'AdminMenu')
            return usermanager.GetUserBookIssued(msgbody)
        else:
            return '-1'
    elif userauth == 'User':
        isBanned = usermanager.GetUserData('Banned', sender)
        if isBanned == 1:
            return 'You are banned.'
        mode = usermanager.GetUserData('Mode', sender)
        if mode == 'GetUserName':
                usermanager.SetUserData('Name', sender, msgbody)
                usermanager.SetUserData('Mode', sender, 'UserMenu')
                return UserMenu()
        if mode == 'UserMenu':
            if msgbody == '1':
                return usermanager.GetUserBookIssued(sender)
            elif msgbody == '2':
                usermanager.SetUserData('Mode', sender, 'ReservationMenu')
                return ReservationMenu()
            elif msgbody == '3':
                return bookmanager.ViewAvailableBooks()
            else:
                return UserMenu()
        elif mode == 'ReservationMenu':
            if msgbody == '1':
                usermanager.SetUserData('Mode', sender, 'ReserveBook')
                return "Enter book ID: "
            elif msgbody == '2':
                usermanager.SetUserData('Mode', sender, 'CancelReserveBook')
                return "Enter book ID: "
            elif msgbody == '3':
                return usermanager.GetUserBookReserved(sender)
            elif msgbody == '4':
                usermanager.SetUserData('Mode', sender, 'UserMenu')
                return UserMenu()
            else:
                return ReservationMenu()       
        elif mode == 'ReserveBook':
            usermanager.SetUserData('Mode', sender, 'ReservationMenu')
            return usermanager.Reserve(msgbody, sender)
        elif mode == 'CancelReserveBook':
            usermanager.SetUserData('Mode', sender, 'ReservationMenu')
            return usermanager.CancelReserve(msgbody, sender)
        else:
            return '-1'
    else:
        usermanager.AddUser([sender, 'NA'])
        usermanager.SetUserData('Mode', sender, 'GetUserName')
        return """
        Hi! It seems this is your first time here.
        Please enter your name.
        """

File: LibraryManager/oldscripts/Controller.java
package com.librarymanagement.demo;

public class Controller {

    public static String Handler(String messageBody, String senderNumber){
        String userauth = Database.getAuth(senderNumber);
        System.out.println(userauth);
        if (userauth.equalsIgnoreCase("Admin")){
            String mode = Database.getData("admindata", "Mode", senderNumber);
            if (mode.equalsIgnoreCase("AdminMenu")){
                if (messageBody.equalsIgnoreCase("1")){
                    return AdminController.issueBookMenu(senderNumber);
                }
                else if (messageBody.equalsIgnoreCase("2")){
                    return AdminController.returnBookMenu(senderNumber);
                }
                else if (messageBody.equalsIgnoreCase("3")){
                    return AdminController.AddBookMenu(senderNumber);
                }
                else if (messageBody.equalsIgnoreCase("4")){
                    return AdminController.RemoveBookMenu(senderNumber);
                }
                else if (messageBody.equalsIgnoreCase("5")){
                    return AdminController.BanUserMenu(senderNumber);
                }
                else if (messageBody.equalsIgnoreCase("6")){
                    return AdminController.unBanUserMenu(senderNumber);
                }
                else if (messageBody.equalsIgnoreCase("7")){
                    return AdminController.addAdminMenu(senderNumber);
                }
                else if (messageBody.equalsIgnoreCase("8")){
                    return AdminController.removeAdminMenu(senderNumber);
                }
                else if (messageBody.equalsIgnoreCase("9")){
                    return AdminController.ViewAllBooks(senderNumber);
                }
                else if (messageBody.equalsIgnoreCase("10")){
                    return AdminController.clearUserDataMenu(senderNumber);
                }
                else{
                    return AdminController.AdminMenu();
                }
            }
            else if (mode.equalsIgnoreCase("AddBookDetails")){
                return AdminController.AddBook(messageBody, senderNumber);
            }
            else if (mode.equalsIgnoreCase("RemoveBook")){
                return AdminController.RemoveBook(messageBody, senderNumber);
            }
            else if (mode.equalsIgnoreCase("BanUser")){
                return AdminController.BanUser(messageBody, senderNumber);
            }
            else if (mode.equalsIgnoreCase("unBanUser")){
                return AdminController.unBanUser(messageBody, senderNumber);
            }
            else if (mode.equalsIgnoreCase("addAdmin")){
                return AdminController.addAdmin(messageBody, senderNumber);
            }
            else if (mode.equalsIgnoreCase("removeAdmin")){
                return AdminController.removeAdmin(messageBody, senderNumber);
            }
            else if (mode.equalsIgnoreCase("clearUser")){
                return AdminController.clearUserData(messageBody, senderNumber);
            }
            else if (mode.equalsIgnoreCase("issueBook")){
                return AdminController.issueBook(messageBody, senderNumber);
            }
            else if (mode.equalsIgnoreCase("returnBook")){
                return AdminController.returnBook(messageBody, senderNumber);
            }
            return "-1";
        }
        else if (userauth.equalsIgnoreCase("User")){
            String isbanned = Database.getData("userdata", "Banned", senderNumber);
            if (isbanned.equalsIgnoreCase("1")){
                return "You are banned.";
            }
            String mode = Database.getData("userdata", "Mode", senderNumber);
            if (mode.equalsIgnoreCase("GetUserName")){
                return UserController.GetUserName(messageBody, senderNumber);
            }
            else if (mode.equalsIgnoreCase("UserMenu")){
                if (messageBody.equalsIgnoreCase("1")){
                    return Database.viewBooksInfo(senderNumber);
                }
                else if (messageBody.equalsIgnoreCase("2")){
                    int suc = Database.setData("userdata", "Mode", senderNumber, "BookReservationMenu");
                    if (suc==-1){
                        return "Could not connect with server. Please try again.";
                    }
                    return UserController.ReservationMenu();
                }
                else if (messageBody.equalsIgnoreCase("3")){
                    return UserController.viewBooks();
                }
                else{
                    return UserController.UserMenu();
                }
            }
            else if (mode.equalsIgnoreCase("BookReservationMenu")){
                if (messageBody.equalsIgnoreCase("1")){
                    return UserController.reserveBookMenu(senderNumber);
                }
                else if (messageBody.equalsIgnoreCase("2")){
                    return UserController.cancelreserveBookMenu(senderNumber);
                }
                else if (messageBody.equalsIgnoreCase("3")){
                    return UserController.viewReservation(senderNumber);
                }
                else{
                    Database.setData("userdata", "Mode", senderNumber, "UserMenu");
                    return UserController.UserMenu();
                }
            }
            else if (mode.equalsIgnoreCase("reserveBook")){
                return UserController.reserveBook(messageBody, senderNumber);
            }
            else if (mode.equalsIgnoreCase("cancelreserveBook")){
                return UserController.cancelreserveBook(messageBody, senderNumber);
            }
            return "-1";
        }
        else{
            int suc = Database.addUserData(senderNumber, "Default", "0", "0", "NA", "NA", "NA", "0", "0", "NA", "NA", "NA", "0", "GetUserName", "0");
            if (suc==-1){
                System.out.println(senderNumber);
                return "Could not connect with server. Please try again.";
            }
            return """
                    Hi! It seems this is your first time here.
                    Please enter your name.
                    """;
        }
    }
}
